sdmetrics/single_table/data_augmentation/utils.py

```diff
@@ -0,0 +1,126 @@
+"""Utils method for data augmentation metrics."""
+
+import pandas as pd
+
+
+def _validate_parameters(
+    real_training_data,
+    synthetic_data,
+    real_validation_data,
+    metadata,
+    prediction_column_name,
+    classifier,
+    fixed_recall_value,
+):
+    """Validate the parameters of the Data Augmentation metrics."""
+    tables = [real_training_data, synthetic_data, real_validation_data]
+    if any(not isinstance(table, pd.DataFrame) for table in tables):
+        raise ValueError(
+            '`real_training_data`, `synthetic_data` and `real_validation_data` must be '
+            'pandas DataFrames.'
+        )
+
+    if not isinstance(metadata, dict):
+        raise TypeError(
+            f"Expected a dictionary but received a '{type(metadata).__name__}' instead."
+            " For SDV metadata objects, please use the 'to_dict' function to convert it"
+            ' to a dictionary.'
+        )
+
+    if not isinstance(prediction_column_name, str):
+        raise TypeError('`prediction_column_name` must be a string.')
+
+    if classifier is not None and not isinstance(classifier, str):
+        raise TypeError('`classifier` must be a string or None.')
+
+    if classifier != 'XGBoost':
```

Comment:
For these types of validations I think it's good to split them into separate smaller functions. The reason for this is that if we decide to expand this class to support more classifiers or create a new class to do that, we might want to overwrite just this part of the validation and keep everything else.